src/otaku.cpp

```diff
@@ -240,7 +240,6 @@ void _update_living_persona(struct char_data *ch, struct obj_data *cyberdeck, in
   ch->persona->decker->mpcp = mpcp;
   ch->persona->decker->hardening = GET_CYBERDECK_HARDENING(cyberdeck);
   ch->persona->decker->response = GET_CYBERDECK_RESPONSE_INCREASE(cyberdeck);
-  ch->persona->decker->io = GET_CYBERDECK_IO_RATING(cyberdeck);
```

Comment:
This should probably be set to _something_. What's the original value?

Reply:
This value is set in `do_connect` (newmatrix.cpp), where it is also capped by the room they're connecting from. The problem was that every time `update_otaku_deck` here is called, it overrides that value and bypasses that limit. The math here is also wrong (it should actually be handled as /10), so otaku are currently downloading files somewhere between 10 and 30 times faster than intended.

This fix is not entirely correct because a change in int (e.g., due to drugs) can cause the living persona's I/O rating to change, but the room limit will pretty much always be the limiting factor. A more correct fix duplicates the code from do_connect.